Documentation/cli-installation-scenarios.md

```diff
@@ -0,0 +1,266 @@
+| OS X                	| Homebrew         	| Though not really "native" as in "available OOB on the system", homebrew has risen as a *de facto* package manager for OS X.                                                                                                                       	|
+| CentOS/RH           	| RPM              	| Though it currently doesn't exist, we should look into enabling RPMs for CentOS/RH-based distributions. This would bring it to par with Debian ones.                                                                                               	|
+
+
+## Installation script
+This approach is a shell one-liner that downloads an installation script and runs it. The installation script will download the latest zip/tarball (depending on the script used) and will unpack it to a given location. After that, the script will print out what needs to be set for the entire CLI to work (env variables, $PATH modification etc.).
+
+This install covers the following main scenario: 
+
+* Local installation on a dev machine
+* Acquiring tools on a CI build
+
+  
+The features the script needs to support/have are:
+* Support for dev and stable channel
+* Support for specifying the version
+* Support for specfying the installation location
+* Support specifying whether the debug package needs to be downloaded
+
+
+### Local installation
+The local installation puts the bits in any folder on disk that the user specifies when invoking the script. The layout is pretty much the same as with native installers.
+
+The local install can "grow up" into a global one simply by pointing the system PATH to the relevant bin folder within the layout on disk. 
+
+Acquiring the tools on the CI server would be done in the same way as on a user machine with a local install: the install script would be invoked with a given set of options and then further build scripts would actually use the tools installed in this way to do their thing (build, restore, pack/publish etc.)
+
+The guidance is, of course, to always use the beta channel for the script, and this is what the script will have as the default option. 
+
+### Installation script features
+The following arguments are needed for the installation script:
+
+>--channel / -Channel
+>Which channel (i.e. "nightly", "preview", "production") to install from; defaults to beta if not present. 
+
+>--version / -Version
+>Which version of CLI to install; you need to specify the version as 3-part version (i.e. 1.0.0-13232323); defaults to latest if not present. 
+
+>--prefix / -InstallDir
+>Path to where to install the CLI bundle; defaults to /.dotnet if not present. The dir is not created if it doesn't exist. 
+
+>--debug / -Debug
+>Download the "fat package" that contains the symbols for debugging the CLI bits; defaults to "false" if not present. 
+
+#### Install the latest nightly CLI
+
+Windows:
+```
+./install.ps1 -Channel nightly
```

Comment:
Ok, I see what you're saying above now with -Channel vs. --channel. Ignore my one-character-versions if desired.